wallet/src/de/schildbach/wallet/service/WalletTransactionMetadataProvider.kt

```
@@ -354,12 +359,16 @@ class WalletTransactionMetadataProvider @Inject constructor(
-                    Fiat.parseFiat(
-                        metadata.currencyCode,
-                        metadata.rate
+                try {
+                    tx.exchangeRate = org.bitcoinj.utils.ExchangeRate(
+                        Fiat.parseFiat(
+                            metadata.currencyCode,
+                            metadata.rate
+                        )
                     )
-                )
+                } catch (e: Exception) {
```

detekt (1.23.7)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)